From b720dd1d1f4a14b12f3fff4517e189a9a891b760 Mon Sep 17 00:00:00 2001
From: pengy <pengy@qti.qualcomm.com>
Date: Thu, 17 Jul 2025 14:05:29 +0800
Subject: [PATCH] systemd: Remove systemd watchdog ping condition

In high IO case, systemd dispatch work may not be done in
watchdog_timeout * 3/4, always ping watchdog unconditionally

Upstream-Status: Inappropriate [Qcom default wdt is 10s]
Signed-off-by: Lei wang <leiwan@qti.qualcomm.com>
---
 src/shared/watchdog.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/shared/watchdog.c b/src/shared/watchdog.c
index 5b113013950..e32b1f0f6b9 100644
--- a/src/shared/watchdog.c
+++ b/src/shared/watchdog.c
@@ -541,8 +541,12 @@ int watchdog_ping(void) {
 
         /* Ping approximately watchdog_timeout/4 after a successful ping, or even less than that
          * after an unsuccessful ping. */
+#if 0
+        /* In high IO case, systemd dispatch work may not be done in
+         * watchdog_timeout * 3/4, always ping watchdog unconditionally */
         if (watchdog_runtime_wait(/* divisor= */ 4) > 0)
                 return 0;
+#endif
 
         return watchdog_ping_now();
 }
-- 
2.34.1

